### DISCUSSION TITLE
How is LLaVA demo run with `--embed` argument set when run via <gradio-app>?

### DISCUSSION THREAD

**Discussion by mattmazzola** [haotian-liu/LLaVA#841] [Q&A] [answered]
I see when the https://llava.hliu.cc/ demo is embedded in the project page https://llava-vl.github.io/ using:

```html
      <gradio-app src="https://llava.hliu.cc"></gradio-app>
```
https://github.com/LLaVA-VL/llava-vl.github.io/blob/a17c5d7ed6db2073040fd581f1bac8cddbbb904b/index.html#L266

Somehow the `--embed` argument is True and thus `build_demo(True)` hides the header and footer.

I didn't understand how this `--embed` argument was passed given the html component above looks standard.
Perhaps this is some native internal flag passed by the component?

Could you link to documentation about this or explain how it works?

**Comment by haotian-liu** [comment]
I used to have an embeded gradio frontend specifically for the project page to hide the header/footer, hosted on a separate domain (e.g. llava-embed.hliu.cc). But this was abandoned back in June, so `--embed` is actually not used in the current demo.

**Comment by mattmazzola** [comment, marked as answer]
Ok, I will rephrase so the answer is clearer for others who may be interested

### Previously 
When the demo was embedded using `<gradio-app>` the header and footer were hidden. 
I saw the [embed_mode](https://github.com/haotian-liu/LLaVA/blob/414cebd318daf563e624ac5d5e02835d40573cb2/llava/serve/gradio_web_server.py#L308C21-L308C21) was used to control this and thought [`--embed`](https://github.com/haotian-liu/LLaVA/blob/414cebd318daf563e624ac5d5e02835d40573cb2/llava/serve/gradio_web_server.py#L455C8-L455C8) must be set somehow.

Haotian clarified that instead of using the embed argument, the `<gradio-app>` was actually referring to a different instance of the demo. (llava-embed.hliu.cc vs llava.hliu.cc)

### Now

Both the full page demo and embedded demo display the header and footer
